packages/opentelemetry-cloud-trace-propagator/test/CloudPropagator.test.ts

```diff
@@ -0,0 +1,171 @@
+        setExtractedSpanContext(Context.ROOT_CONTEXT, spanContext),
+        carrier,
+        defaultSetter
+      );
+
+      assert.deepStrictEqual(
+        carrier[X_CLOUD_TRACE_HEADER],
+        'd4cda95b652f4a1592b449d5929fda1b/7929822056569588882;o=1'
+      );
+    });
+    it('should inject a context of a unsampled span', () => {
+      const spanContext: SpanContext = {
+        traceId: 'd4cda95b652f4a1592b449d5929fda1b',
+        spanId: '6e0c63257de34c92',
+        traceFlags: TraceFlags.NONE,
+      };
+      cloudPropagator.inject(
+        setExtractedSpanContext(Context.ROOT_CONTEXT, spanContext),
+        carrier,
+        defaultSetter
+      );
+      assert.deepStrictEqual(
+        carrier[X_CLOUD_TRACE_HEADER],
+        'd4cda95b652f4a1592b449d5929fda1b/7929822056569588882;o=0'
+      );
+    });
+  });
+
+  describe('.extract()', () => {
+    it('should extract context of a sampled span from carrier', () => {
+      carrier[X_CLOUD_TRACE_HEADER] =
+        'd4cda95b652f4a1592b449d5929fda1b/7929822056569588882;o=1';
+      const extractedSpanContext = getExtractedSpanContext(
+        cloudPropagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
+      );
+
+      assert.deepStrictEqual(extractedSpanContext, {
+        traceId: 'd4cda95b652f4a1592b449d5929fda1b',
+        spanId: '6e0c63257de34c92',
+        traceFlags: TraceFlags.SAMPLED,
+      });
+    });
+
+    it('returns undefined if x-cloud-trace-context header is missing', () => {
+      assert.deepStrictEqual(
+        getExtractedSpanContext(
+          cloudPropagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
+        ),
+        undefined
+      );
+    });
+
+    it('returns undefined if x-cloud-trace-context header is invalid', () => {
+      carrier[X_CLOUD_TRACE_HEADER] = 'invalid!';
+      assert.deepStrictEqual(
+        getExtractedSpanContext(
+          cloudPropagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
+        ),
+        undefined
+      );
+    });
+
+    it('extracts x-cloud-trace-context from list of header', () => {
+      carrier[X_CLOUD_TRACE_HEADER] = [
+        'd4cda95b652f4a1592b449d5929fda1b/7929822056569588882;o=1',
+      ];
+      const extractedSpanContext = getExtractedSpanContext(
+        cloudPropagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
+      );
+      assert.deepStrictEqual(extractedSpanContext, {
+        traceId: 'd4cda95b652f4a1592b449d5929fda1b',
+        spanId: '6e0c63257de34c92',
+        traceFlags: TraceFlags.SAMPLED,
+      });
+    });
+
+    it('should gracefully handle an invalid x-cloud-trace-context header', () => {
+      // A set of test cases with different invalid combinations of a
+      // x-cloud-trace-context header. These should all result in a `undefined`
+      // SpanContext value being extracted.
+
+      const testCases: Record<string, string> = {
+        invalidParts_tooShort: '00-ffffffffffffffffffffffffffffffff',
```

Comment:
These test cases are all invalid because none of them have the required format (they are all missing the slash).

Reply:
Oops my bad, copied from OTEL-js. Updated it now, PTAL